.github/workflows/automatic-cherry-pick.yaml

```diff
@@ -0,0 +1,246 @@
+name: cherry-pick
+
+on:
+  issue_comment:
+    types: [created]
+
+jobs:
+  process_comment:
+    # This job will parse the cherry-pick command and perform several sanity checks.
+    # Ideally we want all pull requests opened from a given command to be correct and successfull, so that
+    # developers don't have to reissue a cherry-pick command for a subset of the commits/branches.
+    # Therefore, we do all possible sanity checks in this job, before any pull request is opened in subsequent jobs.
+    if: github.event.issue.pull_request != '' && contains(github.event.comment.body, 'cherry-pick')
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    outputs:
+      commits: ${{ steps.command.outputs.commits }}
+      branches: ${{ steps.command.outputs.branches }}
+      branch_matrix: ${{ steps.command.outputs.branch_matrix }}
+      status: ${{ steps.report.outputs.status }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          # We need the history of all branches for the sanity checks
+          fetch-depth: 0
+
+      - name: Parse Command
+        # TODO: make this step into a separate job, so that we can handle more than one command per comment
+        id: command
+        shell: bash
+        run: |
+          set -f
+          command_string=$(echo "${{ github.event.comment.body }}" | grep -m 1 cherry-pick)
+          echo "command=$command_string" >> $GITHUB_OUTPUT
+          command=($command_string)
+          command=("${command[@]:1}")
+
+          has_separator=false
+          arg_type=hash
+          commits=()
+          branches=()
+          for token in ${command[@]}; do
+            if [[ "$token" == "into" ]]; then
+              arg_type=branch
+              has_separator=true
+            else
+              if [[ "$arg_type" == "hash" ]]; then
+                commits+=( $token )
+              elif [[ "$arg_type" == "branch" ]]; then
+                branches+=( $token )
+              fi
+            fi
+          done
+
+          # Check command correctness
+          errors=""
+          if [[ "$has_separator" = false ]]; then
+            errors="$errors - the command is missing the \\\`into\\\` separator
+          "
+          fi
+          if [[ -z "$commits" ]]; then
+            errors="$errors - no list of commits to cherry-pick was provided
+          "
+          fi
+          if [[ -z "$branches" ]]; then
+            errors="$errors - no list of target branches was provided
+          "
+          fi
+          if [[ -n "$errors" ]]; then
+            errors="Incorrect cherry-pick command:
+          $errors"
+            printf "ERROR_MSG<<EOF\n%s\nEOF" "$errors" >> $GITHUB_ENV
+          else
```

Comment:
If some of the errors are mutually exclusive, and I think the commits and branches errors are logically dependent on there being a separator, then I'd suggest reflecting that in the logic to make it explicit. Something like this, where the test for separator is logically separate from the tests for branches and commits
```bash
          errors=()
          if [[ "$has_separator" = false ]]; then
            errors+=( "$errors - the command is missing the \\\`into\\\` separator"
          else
            if [[ -z "$commits" ]]; then
              errors+=("$errors - no list of commits to cherry-pick was provided")
            fi
            if [[ -z "$branches" ]]; then
            errors+=( "$errors - no list of target branches was provided )"
            fi
          fi
          if [[ -z "$errors" ]]; then
            errors="Incorrect cherry-pick command:
          ${errors[@]}"
            IFS="\n"; printf "ERROR_MSG<<EOF\n%s\nEOF" "$errors" >> $GITHUB_ENV
          else
```
Note I haven't made that a formal suggestion because it is completely untested. Including for syntax errors (soz).

IMO it would be good to have this whole job as a separate script and add tests for it, but I understand the constraints of time etc.

Reply:
> If some of the errors are mutually exclusive, and I think the commits and branches errors are logically dependent on there being a separator, then I'd suggest reflecting that in the logic to make it explicit.

That's a good idea, as it will make the error message more accurate and easier to understand.

Reply:
Done.